eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/FixImagesVarDirCommand.php

```diff
@@ -0,0 +1,355 @@
+
+    /**
+     * @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler
+     */
+    private $db;
+
+    /**
+     * @var \eZ\Publish\Core\Persistence\Legacy\Content\Gateway
+     */
+    private $contentGateway;
+
+    /**
+     * @var \eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway
+     */
+    private $imageGateway;
+
+    /**
+     * @var \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ChainConfigResolver
+     */
+    private $configResolver;
+
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\SiteAccess
+     */
+    private $siteaccess;
+
+    /**
+     * @var int
+     */
+    protected $done = 0;
+
+    /**
+     * @var string
+     */
+    private $phpPath;
+
+    /**
+     * @var bool
+     */
+    private $dryRun;
+
+    /**
+     * @var int
+     */
+    private $varDir;
+
+    /**
+     * @var array
+     */
+    private $imageAttributes = [];
+
+    /**
+     * @param \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ChainConfigResolver; $configResolver
+     * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $db
+     * @param \eZ\Publish\Core\MVC\Symfony\SiteAccess $siteaccess
+     * @param \eZ\Publish\Core\Persistence\Legacy\Content\Gateway $contentGateway
+     * @param \eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway $imageGateway
+     */
+    public function __construct(
+        ChainConfigResolver $configResolver,
+        DatabaseHandler $db,
+        SiteAccess $siteaccess,
+        ContentGateway $contentGateway,
+        ImageGateway $imageGateway
+    ) {
+        parent::__construct();
+        $this->db = $db;
+        $this->configResolver = $configResolver;
+        $this->siteaccess = $siteaccess;
+        $this->contentGateway = $contentGateway;
+        $this->imageGateway = $imageGateway;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:fix_images_var_dir')
+            ->setDescription(
+                'This update script will fix references to images that are not placed in the current var_dir.'
+            )
+            ->addOption(
+                'dry-run',
+                null,
+                InputOption::VALUE_NONE,
+                'Execute a dry run'
+            )
+            ->addOption(
+                'iteration-count',
+                null,
+                InputArgument::OPTIONAL,
+                'Limit how many records get updated by single process',
+                self::DEFAULT_ITERATION_COUNT
+            )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> fixes references to images that are not placed in the current var_dir.
+
+This may for instance occur when the var_dir setting is changed. This script will rename the files, and update the
+database references to the new path
+
+Since this script can potentially run for a very long time, to avoid memory exhaustion run it in
```

Comment:
On prod env will always be prod. It is probably better suggest to use --no-debug for non-prod envs